src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,156 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> newNode = new Node(null, value, null);
+        if (isEmpty()) {
+            first = last = newNode;
+            size++;
+            return true;
+        }
+        last.next = newNode;
+        newNode.prev = last;
+        last = newNode;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == index) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        Node<T> oldNode = getNode(index);
+        Node<T> newNode = new Node<T>(oldNode.prev, value, oldNode);
+        if (index == 0) {
+            first = newNode;
+        }
+        if (oldNode.prev != null) {
+            oldNode.prev.next = newNode;
+            oldNode.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T elem : list) {
+            add(elem);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> result;
+        if (index < size / 2) {
+            result = first;
+            for (int i = 0; i < index; i++) {
+                result = result.next;
+            }
+        } else {
+            result = last;
+            for (int i = size - 1; i > index; i--) {
+                result = result.prev;
+            }
+        }
+        return (T) result.element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = getNode(index);
+        T removedValue = node.element;
+        node.element = value;
+        return removedValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = getNode(index);
+        if (node.prev != null) {
+            node.prev.next = node.next;
```

Comment:
it is difficult to understand
let's create local variable `prev` and `next`

Reply:
done